### PR TITLE
Final ca

### PR DESCRIPTION
**Pull request by 1004gun1**
fixed two issues: 
1. the profile view appears after the game finishes
2. click the cross button three times to close the profile view.

### CODE REVIEW

File: src/main/java/interface_adapter/history/HistoryPresenter.java [hunk 1]
```diff
@@ -28,12 +28,25 @@ public void prepareSuccessView(HistoryOutputData historyOutputData) {
...
-        profileViewModel.firePropertyChange("losses", null, loss);
-        profileViewModel.firePropertyChange("status", null, state);
 
         gameEndViewModel.firePropertyChange("wins", null, win);
         gameEndViewModel.firePropertyChange("losses", null, loss);
         gameEndViewModel.firePropertyChange("winRate", null, winrates);
     }
+
+    @Override
+    public void updatestatus(HistoryOutputData historyOutputData) {
```
Comment (RaghavAtRuntime) [inline src/main/java/interface_adapter/history/HistoryPresenter.java]: Refactor to updateStatus() to follow naming conventions

File: src/main/java/use_case/History/HistoryInteractor.java [hunk 2]
```diff
@@ -14,13 +14,28 @@ public HistoryInteractor(UserService userService, HistoryOutputBoundary historyP
...
+    }
+
+
+    @Override
+    public void updatestatus(HistoryInputData historyInputData) {
+
         String name = historyInputData.getUsername();
         int win = userService.getUserWins(name);
         int loss = userService.getUserLosses(name);
         String state = historyInputData.getState();
```
Comment (RaghavAtRuntime) [inline src/main/java/use_case/History/HistoryInteractor.java]: Refactor to status for clarity

Comment (RaghavAtRuntime) [reply src/main/java/use_case/History/HistoryInteractor.java]: Also refactor getState to getStatus

File: src/main/java/interface_adapter/history/HistoryPresenter.java [hunk 3]
```diff
@@ -28,12 +28,25 @@ public void prepareSuccessView(HistoryOutputData historyOutputData) {
...
+
+        int win = historyOutputData.getWin();
+        int loss = historyOutputData.getLoss();
+        String state = historyOutputData.getState();
```
Comment (RaghavAtRuntime) [inline src/main/java/interface_adapter/history/HistoryPresenter.java]: refactor to getStatus()

File: src/main/java/interface_adapter/history/HistoryPresenter.java [hunk 4]
```diff
@@ -28,12 +28,25 @@ public void prepareSuccessView(HistoryOutputData historyOutputData) {
...
+        final ProfileState currentstate = profileViewModel.getState();
+        currentstate.setState(state);
```
Comment (RaghavAtRuntime) [inline src/main/java/interface_adapter/history/HistoryPresenter.java]: refactor to setStatus()

**Review by RaghavAtRuntime — CHANGES_REQUESTED** [review]

**Review by ibrmaj — APPROVED** [review]

**Review by ibrmaj — DISMISSED** [review]

**Review by RaghavAtRuntime — DISMISSED** [review]

**Pull request merged**